picard/ui/itemviews/basetreeview.py

```diff
@@ -424,7 +426,8 @@ def add_actions(*args):
                         action_menu = plugin_menus[key] = action_menu.addMenu(key[-1])
                 action_menu.addAction(action)
 
-        if scripts:
+        if has_tagging_scripts(config=config):
+            scripts = iter_tagging_scripts_from_config(config=config)
```

Comment:
Not sure about that. We now have to read `config.setting['list_of_scripts']` twice, and config access is kind of expensive. Doesn't probably matter much in this function, but still.

What about using `iter_tagging_scripts_from_tuples` here instead?

```python
scripts = config.setting['list_of_scripts']
if scripts:
    scripts_menu = ScriptsMenu(iter_tagging_scripts_from_tuples(scripts), _("&Run scripts"), menu)
    ....
```

Reply:
The idea was to regroup calls to `config.setting['list_of_scripts']` in one place, but I agree that's not optimal, still WIP

Reply:
Done and rebased